### PR TITLE
fix skipping

### PR DESCRIPTION
**Pull request by lipovsek-aws**
Replace filtering and returning with explicit skip.

### CODE REVIEW

File: axlearn/common/host_array_test.py [hunk 1]
```diff
@@ -28,26 +28,24 @@ def is_supported(
...
     @parameterized.parameters(
-        filter(
-            lambda params: is_supported(*params),
-            itertools.product(
-                ("cpu", "tpu"),  # platform,
-                ((1, 1), (4, 1), (2, 2), (8, 1), (4, 2)),  # mesh_shape
-                (1, 16),  # global_batch_size
-                (DataPartitionType.FULL, DataPartitionType.REPLICATED),  # data_partition
-            ),
+        itertools.product(
```
Comment (patrick-toulme) [inline axlearn/common/host_array_test.py:44]: Why is this change required?

File: axlearn/common/inference_test.py [hunk 2]
```diff
@@ -292,17 +295,14 @@ def init_state(prng_key):
...
-            itertools.product(
-                ("cpu", "gpu", "tpu"),  # platform,
-                ((1, 1), (4, 1), (2, 2), (8, 1), (4, 2)),  # mesh_shape
-                (jnp.float32, jnp.bfloat16),  # param_dtype
-                (None, jnp.float32, jnp.bfloat16),  # inference_dtype
-                (1, 16),  # global_batch_size
-                (DataPartitionType.FULL, DataPartitionType.REPLICATED),  # data_partition
-                (True, False),  # whether use ema weight
-            ),
+        itertools.product(
```
Comment (patrick-toulme) [inline axlearn/common/inference_test.py:298]: Why is this change required?

**Review by patrick-toulme — COMMENTED** [review]
Looks good. Couple comments.

File: axlearn/common/host_array_test.py (hunk 1 above)
Comment (lipovsek-aws) [reply axlearn/common/host_array_test.py:44]: instead of filtering in @parameterized.parameters (which means tests are not skipped but disappear and can result in pytest error) I'm doing pytest.skip in the function (line 65) which won't result in error or disappearing (but skip instead)

File: axlearn/common/inference_test.py (hunk 2 above)
Comment (lipovsek-aws) [reply axlearn/common/inference_test.py:298]: instead of filtering in @parameterized.parameters (which means tests are not skipped but disappear and can result in pytest error) I'm doing pytest.skip in the function (line 326)

**Comment by patrick-toulme** [comment]
Thanks!
